src/lib/contracts/common/interfaces/ITokenContract.ts

```diff
@@ -2,6 +2,11 @@ import type { TypedContractMethod } from "@/lib/contracts/generated/common";
 import type { BaseContract, AddressLike } from "ethers";
 
 export interface ITokenContract extends BaseContract {
+  allowance: TypedContractMethod<
```

Comment:
is the point of these to just add the the functions that we use in the app so that way we can ensure the contracts we are using have the needed properties? bc these are surely not all of the functions on the contract.

another benefit to viem/wagmi is it typechecks against the abi so you don't have to do this
I feel like a painpoint here is you need to remember to update this file everything you add a block of code that calls a function not yet included here